src/components/ShoppingList/ShoppingList.js

```diff
@@ -11,13 +13,41 @@ function ShoppingList({ listId }) {
     db.collection(`lists/${listId}/items`).orderBy('purchaseInterval', 'asc'),
   );
 
-  const handleCheck = (e) => {
-    const itemId = e.target.value;
+  // Helper function to get the latest interval between purchases (expects Luxon date objects)
+  const getLatestInterval = ({ lastPurchaseDate, newPurchaseDate }) => {
+    const duration = newPurchaseDate.diff(lastPurchaseDate, ['days']);
+    return Math.round(duration.as('days'));
+  };
+
+  const checkAsPurchased = (itemId, item) => {
+    // convert lastPurchaseDate from firestore and JS current time to Luxon DateTime objects
+    const lastPurchaseDate = item.lastPurchaseDate?.seconds
+      ? DateTime.fromSeconds(item.lastPurchaseDate.seconds)
+      : null; // for new items, lastPurchaseDate will be null so keep it null
+    const newPurchaseDate = DateTime.fromSeconds(Math.floor(Date.now() / 1000));
+
+    // if lastPurchaseDate is null (item not yet purchased), a latest interval can't be
+    // calculated, so in that case set latestInterval to the current purchaseInterval
+    const latestInterval =
+      lastPurchaseDate === null
+        ? item.purchaseInterval
+        : getLatestInterval({ lastPurchaseDate, newPurchaseDate });
+
+    const newPurchaseInterval = calculateEstimate(
+      item.purchaseInterval,
+      latestInterval,
+      item.numberOfPurchases,
+    );
+
```

Comment:
This was a really helpful update, especially with the updated comments, great job!

Reply:
👏🏽